be/src/exprs/jit/jit_expr.cpp

```diff
@@ -0,0 +1,151 @@
+
+    FunctionContext::TypeDesc return_type = AnyValUtil::column_type_to_type_desc(_type);
+    std::vector<FunctionContext::TypeDesc> args_types;
+
+    for (Expr* child : _children) {
+        args_types.push_back(AnyValUtil::column_type_to_type_desc(child->type()));
+    }
+    _fn_context_index = context->register_func(state, return_type, args_types);
+
+    return Status::OK();
+}
+
+Status JITExpr::open(RuntimeState* state, ExprContext* context, FunctionContext::FunctionStateScope scope) {
+    RETURN_IF_ERROR(Expr::open(state, context, scope));
+
+    FunctionContext* fn_ctx = context->fn_context(_fn_context_index);
+
+    // TODO(Yueyang): remove this time cost.
+    auto start = std::chrono::high_resolution_clock::now();
+
+    // Compile the expression into native code and retrieve the function pointer.
+    auto* jit_wapper = JITWapper::get_instance();
+    if (!jit_wapper->initialized()) {
+        return Status::JitCompileError("JIT is not supported");
+    }
+
+    auto function = jit_wapper->compile_scalar_function(context, _expr);
+
+    auto end = std::chrono::high_resolution_clock::now();
+    std::chrono::duration<double> elapsed = end - start;
+    if (!function.ok()) {
+        LOG(INFO) << "JIT: JIT compile failed, time cost: " << elapsed.count() << " s"
+                  << " Reason: " << function.status();
+    } else {
+        LOG(INFO) << "JIT: JIT compile success, time cost: " << elapsed.count() << " s";
+    }
+
+    auto function_state = _pool->add(new JITFunctionState{function.value_or(nullptr)});
+
+    // TODO(Yueyang): check THREAD_LOCAL or FRAGMENT_LOCAL.
+    fn_ctx->set_function_state(FunctionContext::FunctionStateScope::THREAD_LOCAL, function_state);
+
+    return Status::OK();
+}
+
+StatusOr<ColumnPtr> JITExpr::evaluate_checked(starrocks::ExprContext* context, Chunk* ptr) {
+    FunctionContext* fn_ctx = context->fn_context(_fn_context_index);
+    auto state = reinterpret_cast<JITFunctionState*>(fn_ctx->get_function_state(FunctionContext::THREAD_LOCAL));
+    // If the expr fails to compile, evaluate using the original expr.
+    if (UNLIKELY(state->_function == nullptr)) {
+        LOG(ERROR) << "JIT: JIT compile failed, fallback to original expr";
+        // TODO(Yueyang): fallback to original expr.
+        return _expr->evaluate_checked(context, ptr);
+    }
+
+    Columns args;
+    args.reserve(_children.size() + 1);
+    for (Expr* child : _children) {
+        ColumnPtr column = EVALUATE_NULL_IF_ERROR(context, child, ptr);
+        args.emplace_back(column);
+    }
+
+#ifndef NDEBUG
+    if (ptr != nullptr) {
+        size_t size = ptr->num_rows();
+        // Ensure all columns have the same size
+        for (const ColumnPtr& c : args) {
+            CHECK_EQ(size, c->size());
+        }
+    }
+#endif
+
+    for (const auto& column : args) {
+        if (column->only_null()) {
+            return ColumnHelper::create_const_null_column(column->size());
+        }
```

Comment:
seems not true? if we support if null, or is null, we shouldn't short curt here.

Reply:
Comments have been added, will be modified in subsequent PR.